ecospheres_migrator/geonetwork.py

```diff
@@ -156,6 +157,38 @@ def get_record(self, uuid: str) -> etree._ElementTree:
         r.raise_for_status()
         return etree.fromstring(r.content, parser=None)
 
+    def _extract_uuid_from_put_response(self, payload: dict):
+        """
+        Create record UUID is not in the `uuid` but in `metadatasInfos`:
+        ```
+        "metadataInfos":{
+            "259":[
+                {
+                    "message":"Metadata imported from XML with UUID '7d447744-1be5-4be0-8b46-6be0d36ec90f'",
+                    "date":"2024-09-12T15:39:41"
+                }
+            ]
+        },
+        """
+        metadata_infos = payload.get("metadataInfos")
+        if not metadata_infos:
+            return None
+
+        uuid_match = None
+        for md_info in metadata_infos.values():
+            for info in md_info:
+                message = info.get("message")
+                uuid_match = re.search(
+                    r"'([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'", message
+                )
+                if uuid_match:
+                    uuid_match = uuid_match.group(1)
```

Comment:
Return here instead of the several breaks? (also avoids reassigning/transtyping `uuid_match`).